linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -956,25 +923,19 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
       }
 
       /** Look up the target of a static call to an instance method. */
-      protected def staticCall(className: String,
+      protected def staticCall(className: String, namespace: MemberNamespace,
           methodName: String): Option[MethodID] = {
-        classes.get(className).fold {
-          // If it's not a class, it must be a call to a default intf method
-          val defaultsNS = defaults(className)
-          MethodImpl.this.registerStaticCall(defaultsNS.myInterface, methodName)
-          defaultsNS.methods.get(methodName)
-        } { clazz =>
-          MethodImpl.this.registerStaticCall(clazz.myInterface, methodName)
-          clazz.lookupMethod(methodName)
-        }
-      }
 
-      /** Look up the target of a call to a static method. */
-      protected def callStatic(className: String,
-          methodName: String): Option[MethodID] = {
-        val staticsNS = statics(className)
-        registerCallStatic(staticsNS.myInterface, methodName)
-        staticsNS.methods.get(methodName)
+        def inStaticsLike =
+          staticLikes(className)(namespace.ordinal)
+
+        val container =
+          if (namespace != MemberNamespace.Public) inStaticsLike
+          else classes.get(className).getOrElse(inStaticsLike)
```

Comment:
This would also benefit from typing `staticLikes` with `MethodContainer` only (and of course renaming it).